lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -871,12 +871,23 @@ def _handle_job_failure(self, job, job_state):
                 self.__cleanup_k8s_guest_ports(job_state.job_wrapper, job)
             self.__cleanup_k8s_job(job)
         except Exception:
-            log.exception("Could not clean up k8s batch job. Ignoring...")
+            log.exception("Could not clean up a failed k8s batch job. Ignoring...")
         return mark_failed
 
-    def __cleanup_k8s_job(self, job):
+    def __cleanup_k8s_job(self, job, retry=5, timeout=10):
+        log.debug(f"Cleaning up job with K8s id {job.name}.")
         k8s_cleanup_job = self.runner_params["k8s_cleanup_job"]
-        delete_job(job, k8s_cleanup_job)
+        for i in range(retry):
+            try:
+                delete_job(job, k8s_cleanup_job)
+                break
+            except Exception as exc:
+                if i == retry - 1:
+                    raise exc
+                log.warning(
+                    f"Failed to delete job {job.name} (attempt {i + 1}/{retry}). Retrying deletion in {timeout} seconds."
+                )
+                time.sleep(timeout)
```

Comment:
We probably need to avoid blocking the handler thread with a timeout here, so something like `self.work_queue.put((self.__cleanup_k8s_job, job)) would cause the cleanup task to be rescheduled. We'd have to add some logic for limiting the number of retries as well.